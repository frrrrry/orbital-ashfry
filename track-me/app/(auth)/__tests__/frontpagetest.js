
import React from 'react';
import FrontPage from "../frontpage"; 
import { render } from '@testing-library/react-native';

describe('welcome screen', () => {
    /*
    it('renders default elements', () => {
        render(<FrontPage />);
    })
    */
    it ('should return false', () => {
      expect(false).toBeTruthy(); 
    })

    it('should navigate to login page when sign in button pressed', async () => {
        // given
        const history = createMemoryHistory();
      
        // when
        render(<Router history={history}><FrontPage /></Router>);
      
        // then
        await waitFor(() => {
          expect(history.location.pathname).toBe('../login');
        });
      });

    // should navigate to register page
    it('should navigate to sign up page when sign up button pressed', async () => {
        // given
        const history = createMemoryHistory();
      
        // when
        render(<Router history={history}><FrontPage /></Router>);
      
        // then
        await waitFor(() => {
          expect(history.location.pathname).toBe('../signup');
        });
    });
})
