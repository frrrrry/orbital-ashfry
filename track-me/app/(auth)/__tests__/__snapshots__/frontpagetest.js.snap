// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`welcome screen renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    App
  </Text>
</View>
`;
